airflow/cli/commands/celery_command.py

```diff
@@ -115,23 +127,19 @@ def worker(args):
         'autoscale': autoscale,
         'hostname': args.celery_hostname,
         'loglevel': conf.get('logging', 'LOGGING_LEVEL'),
+        'pidfile': pid_file_path,
     }
 
     if conf.has_option("celery", "pool"):
         options["pool"] = conf.get("celery", "pool")
 
     if args.daemon:
-        pid, stdout, stderr, log_file = setup_locations("worker",
-                                                        args.pid,
-                                                        args.stdout,
-                                                        args.stderr,
-                                                        args.log_file)
+        # Run Celery worker as daemon
         handle = setup_logging(log_file)
         stdout = open(stdout, 'w+')
         stderr = open(stderr, 'w+')
 
         ctx = daemon.DaemonContext(
-            pidfile=TimeoutPIDLockFile(pid, -1),
```

Comment:
Also, it seems that -D flag doesn't work with this line. No error is propagated and I see no new celery processes. When I remove it, everything works as expected?  WDYT? @potiuk @mik-laj @dimberman

Reply:
I'd have to discuss it in detail in-person, I am not 100% sure how PID handling should be done now :). Seems we have few places where it happens and I am not sure I understand all the paths.

Reply:
After my changes pid file is handled by Celery builtin mechanism and standard lockfile / psutil methods in stop command.

Reply:
After all this works fine, the problem is when celery process exits unexpectedly and pid file is not removed. Then running worker with -D won't start up celery and user will not receive any information.